app/src/main/java/com/google/wear/whereami/data/LocationViewModel.kt

```diff
@@ -14,33 +14,63 @@
 package com.google.wear.whereami.data
 
 import android.Manifest
-import android.content.ComponentName
 import android.content.Context
 import android.content.pm.PackageManager
-import androidx.wear.complications.datasource.ComplicationDataSourceUpdateRequester
+import com.dropbox.android.external.store4.Fetcher
+import com.dropbox.android.external.store4.MemoryPolicy
+import com.dropbox.android.external.store4.Store
+import com.dropbox.android.external.store4.StoreBuilder
 import com.google.android.gms.location.LocationRequest
 import com.patloew.colocation.CoGeocoder
 import com.patloew.colocation.CoLocation
+import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.FlowPreview
 import kotlinx.coroutines.withContext
+import java.io.IOException
+import kotlin.time.Duration
 
-class LocationViewModel(val applicationContext: Context) {
+class LocationViewModel(private val applicationContext: Context, private val coroutineScope: CoroutineScope) {
     private val coGeocoder = CoGeocoder.from(applicationContext)
     private val coLocation = CoLocation.from(applicationContext)
 
-    suspend fun readLocationResult(): LocationResult {
+    @OptIn(FlowPreview::class, kotlinx.coroutines.ExperimentalCoroutinesApi::class,
+        kotlin.time.ExperimentalTime::class
+    )
+    val store: Store<String, LocationResult> = StoreBuilder
+        .from<String, LocationResult>(Fetcher.of { readLocationResult() })
```

Comment:
Is this also caching errors? I think we'd want to retry as soon as possible if the last request was an error, right? So we'd probably only want to cache successful results?

Reply:
I was treating errors as results.  I think I want something fairly nuanced here, like some comparative ordering.

fresh result
recent result
fresh error
stale result
recent error
stale error (don't store)

We definitely don't want to constantly poll for new locations when there is an error.

Reply:
I guess it depends on the error. From what I've seen, a lot of errors on Wear are connection errors, or GPS location errors, that often can be fixed by a retry. If it was an error from the server (e.g. that location doesn't exist, then yeah we probably don't need to retry that).

But generally, I think if there is an error, you don't want to cache it (instead you'd want to re-use the old value and just let the user know that it is stale via the "time ago" text.) That way, they can manually open up the activity from the complication or tile to force a refresh. 

If we cache errors, that means they won't be able to force a refresh by opening the activity. (Unless there's some kind of UI affordance to do that, like a manual refresh button). But I think it would be a better user experience if by the time you got to the activity, it would be retrying for the user automatically instead of making them press the button.

Reply:
Sounds good.  Mind if I follow up on that, I'll put that logic in a couple of places

1) the stream of updates, to drop a following update that is "worse" than the current.
2) A JobService that is causing refresh based on whatever signals are available e.g. inertia or footstep counts.